edx_django_utils/hooks/pipeline.py

```diff
@@ -0,0 +1,83 @@
+"""
+TODO: add description
+"""
+from logging import getLogger
+
+from .exceptions import HookException
+from .utils import get_functions_for_pipeline, get_pipeline_configuration
+
+log = getLogger(__name__)
+
+
+def run_filter_pipeline(trigger_name, *args, **kwargs):
+    """
+    Function that manages the filters pipeline.
+
+    Arguments:
+        trigger_name:
+
+    Returns:
+        result:
+    """
+    pipeline, is_async = get_pipeline_configuration(trigger_name)
+
+    if is_async:
+        result = _run_pipeline(pipeline, *args, **kwargs)  # TODO: change to async call.
+    else:
+        result = _run_pipeline(pipeline, *args, **kwargs)
+
+    return result
+
+
+def run_action_pipeline(trigger_name, *args, **kwargs):
+    """
+    Function that manages the filters pipeline.
+
+    Arguments:
+        trigger_name:
+
+    Returns:
+        None:
+    """
+    pipeline, is_async = get_pipeline_configuration(trigger_name)
+
+    if is_async:
+        _run_pipeline(pipeline, *args, **kwargs)  # TODO: change to async call.
+    else:
+        _run_pipeline(pipeline, *args, **kwargs)
+
+
+def _run_pipeline(pipeline, *args, **kwargs):
+    """
+    Will check in the django setting HOOKS_EXTENSIONS the trigger_name key and call their configured
+    functions.
+
+    Arguments:
+        trigger_name: a string that determines which is the trigger of this pipeline.
+
+    Returns:
+        out: dictionary with results of each trigger.  # TODO: add type to docstrings
+    """
+    triggers = get_functions_for_pipeline(pipeline)
+
+    out = kwargs.copy()
+    for trigger in triggers:
+        try:
+            result = trigger(*args, **out) or {}
+            if not isinstance(result, dict):
+                return result
+            out.update(result)
+        except HookException as exc:
+            raise exc
```

Comment:
I think that we have to consider the case in which we are in an action (this should not be raised in those cases).